app/services/service_agreement_creator.rb

```diff
@@ -22,22 +24,22 @@ def call(service_agreement_params)
   private
 
   def modify_org_service_agreement_params(service_agreement_params)
-    civil_servant = civil_servant(service_agreement_params)
+    @civil_servant = find_or_create_civil_servant(service_agreement_params)
 
-    modified_params = service_agreement_params.merge(organization_agreed: true, civil_servant: civil_servant)
-    modified_params.delete(:civil_servant_attributes) if civil_servant.persisted?
+    modified_params = service_agreement_params.merge(organization_agreed: true, civil_servant: @civil_servant)
+    modified_params.delete(:civil_servant_attributes) if @civil_servant.persisted?
     modified_params
   end
 
-  def civil_servant(service_agreement_params)
-    @civil_servant ||= find_civil_servant_by_email(service_agreement_params) || CivilServant.new(user: User.new)
+  def civil_servant_email_from_params(params)
+    params.dig(:civil_servant_attributes, :user_attributes, :email)
   end
 
-  def find_civil_servant_by_email(service_agreement_params)
-    CivilServant.joins(:user).find_by(
-      users: {
-        email: service_agreement_params.dig(:civil_servant_attributes, :user_attributes, :email)
-      }
-    )
+  def find_or_create_civil_servant(email = nil)
+    @civil_servant || find_civil_servant_by_email(email) || CivilServant.new(user: User.new)
+  end
```

Comment:
Why is this called find or create if it does not create a civil servant?
```suggestion
  def find_or_initialize_civil_servant(email = nil)
    @civil_servant || find_civil_servant_by_email(email) || CivilServant.new(user: User.new)
  end
```